actions/retweet.py

```diff
@@ -0,0 +1,70 @@
+from twitter import Twitter
+from twitter import OAuth
+
+import requests
+import os
+from io import StringIO
+
+from st2common.runners.base_action import Action
+
+# 1 MB chunks
+DOWNLOAD_CHUNK_SIZE_BYTES = 1 * 1024 * 1024
+
+__all__ = [
+    'UpdateStatusAction'
+]
+
+
+class UpdateStatusAction(Action):
+
+    def download_url(self, url):
+        data_buffer = StringIO()
+        req = requests.get(url, stream=True)
+        for chunk in req.iter_content(chunk_size=DOWNLOAD_CHUNK_SIZE_BYTES):
+            data_buffer.write(chunk)
+        return data_buffer.getvalue()
+
+    def read_path_or_url(self, path_or_url):
+        # If path_or_url is a URL, then download the file
+        # Else create a file object for the given path
+        if path_or_url.startswith('http'):
+            data = self.download_url(path_or_url)
+        else:
+            with open(os.path.realpath(path_or_url), 'rb') as f:
+                data = f.read()
+        return data
+
+    def run(self, tweet_id, accounts=[]):
+
+        rt_accounts = []
+
+        # Iterate over provided list of accounts to retweet from
+        for account in accounts:
```

Comment:
looks like this is copy/pasted from `get_twitter_tokens()` anyway to reuse that function?